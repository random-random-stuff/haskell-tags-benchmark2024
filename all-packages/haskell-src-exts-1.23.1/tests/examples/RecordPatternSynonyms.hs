{-# LANGUAGE PatternSynonyms #-}
module ShouldCompile where

pattern Single{x} = [x]

pattern Double{y,z} = (y,z)

pattern More{x} <- (x,_) where
  More x = (x, Nothing)

-- Selector
selector :: Int
selector = x [5]

update :: [String]
update = ["String"] { x = "updated" }
