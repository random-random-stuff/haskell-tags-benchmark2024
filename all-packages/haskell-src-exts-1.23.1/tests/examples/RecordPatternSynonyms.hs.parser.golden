ParseOk
  ( Module
      SrcSpanInfo
        { srcInfoSpan =
            SrcSpan "tests/examples/RecordPatternSynonyms.hs" 1 1 17 1
        , srcInfoPoints =
            [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 1 1 1 1
            , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 2 1 2 1
            , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 2 1 2 1
            , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 1 4 1
            , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 1 6 1
            , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 1 8 1
            , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 12 1 12 1
            , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 1 13 1
            , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 15 1 15 1
            , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 1 16 1
            , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 17 1 17 1
            , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 17 1 17 1
            ]
        }
      (Just
         (ModuleHead
            SrcSpanInfo
              { srcInfoSpan =
                  SrcSpan "tests/examples/RecordPatternSynonyms.hs" 2 1 2 27
              , srcInfoPoints =
                  [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 2 1 2 7
                  , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 2 22 2 27
                  ]
              }
            (ModuleName
               SrcSpanInfo
                 { srcInfoSpan =
                     SrcSpan "tests/examples/RecordPatternSynonyms.hs" 2 8 2 21
                 , srcInfoPoints = []
                 }
               "ShouldCompile")
            Nothing
            Nothing))
      [ LanguagePragma
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/RecordPatternSynonyms.hs" 1 1 1 33
            , srcInfoPoints =
                [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 1 1 1 13
                , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 1 30 1 33
                ]
            }
          [ Ident
              SrcSpanInfo
                { srcInfoSpan =
                    SrcSpan "tests/examples/RecordPatternSynonyms.hs" 1 14 1 29
                , srcInfoPoints = []
                }
              "PatternSynonyms"
          ]
      ]
      []
      [ PatSyn
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 1 4 24
            , srcInfoPoints =
                [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 1 4 8
                , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 19 4 20
                ]
            }
          (PRec
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 9 4 18
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 15 4 16
                   , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 17 4 18
                   ]
               }
             (UnQual
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 9 4 15
                  , srcInfoPoints = []
                  }
                (Ident
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 9 4 15
                     , srcInfoPoints = []
                     }
                   "Single"))
             [ PFieldPun
                 SrcSpanInfo
                   { srcInfoSpan =
                       SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 16 4 17
                   , srcInfoPoints = []
                   }
                 (UnQual
                    SrcSpanInfo
                      { srcInfoSpan =
                          SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 16 4 17
                      , srcInfoPoints = []
                      }
                    (Ident
                       SrcSpanInfo
                         { srcInfoSpan =
                             SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 16 4 17
                         , srcInfoPoints = []
                         }
                       "x"))
             ])
          (PList
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 21 4 24
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 21 4 22
                   , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 23 4 24
                   ]
               }
             [ PVar
                 SrcSpanInfo
                   { srcInfoSpan =
                       SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 22 4 23
                   , srcInfoPoints = []
                   }
                 (Ident
                    SrcSpanInfo
                      { srcInfoSpan =
                          SrcSpan "tests/examples/RecordPatternSynonyms.hs" 4 22 4 23
                      , srcInfoPoints = []
                      }
                    "x")
             ])
          ImplicitBidirectional
      , PatSyn
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 1 6 28
            , srcInfoPoints =
                [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 1 6 8
                , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 21 6 22
                ]
            }
          (PRec
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 9 6 20
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 15 6 16
                   , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 17 6 18
                   , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 19 6 20
                   ]
               }
             (UnQual
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 9 6 15
                  , srcInfoPoints = []
                  }
                (Ident
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 9 6 15
                     , srcInfoPoints = []
                     }
                   "Double"))
             [ PFieldPun
                 SrcSpanInfo
                   { srcInfoSpan =
                       SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 16 6 17
                   , srcInfoPoints = []
                   }
                 (UnQual
                    SrcSpanInfo
                      { srcInfoSpan =
                          SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 16 6 17
                      , srcInfoPoints = []
                      }
                    (Ident
                       SrcSpanInfo
                         { srcInfoSpan =
                             SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 16 6 17
                         , srcInfoPoints = []
                         }
                       "y"))
             , PFieldPun
                 SrcSpanInfo
                   { srcInfoSpan =
                       SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 18 6 19
                   , srcInfoPoints = []
                   }
                 (UnQual
                    SrcSpanInfo
                      { srcInfoSpan =
                          SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 18 6 19
                      , srcInfoPoints = []
                      }
                    (Ident
                       SrcSpanInfo
                         { srcInfoSpan =
                             SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 18 6 19
                         , srcInfoPoints = []
                         }
                       "z"))
             ])
          (PTuple
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 23 6 28
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 23 6 24
                   , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 25 6 26
                   , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 27 6 28
                   ]
               }
             Boxed
             [ PVar
                 SrcSpanInfo
                   { srcInfoSpan =
                       SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 24 6 25
                   , srcInfoPoints = []
                   }
                 (Ident
                    SrcSpanInfo
                      { srcInfoSpan =
                          SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 24 6 25
                      , srcInfoPoints = []
                      }
                    "y")
             , PVar
                 SrcSpanInfo
                   { srcInfoSpan =
                       SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 26 6 27
                   , srcInfoPoints = []
                   }
                 (Ident
                    SrcSpanInfo
                      { srcInfoSpan =
                          SrcSpan "tests/examples/RecordPatternSynonyms.hs" 6 26 6 27
                      , srcInfoPoints = []
                      }
                    "z")
             ])
          ImplicitBidirectional
      , PatSyn
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 1 8 25
            , srcInfoPoints =
                [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 1 8 8
                , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 17 8 19
                ]
            }
          (PRec
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 9 8 16
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 13 8 14
                   , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 15 8 16
                   ]
               }
             (UnQual
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 9 8 13
                  , srcInfoPoints = []
                  }
                (Ident
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 9 8 13
                     , srcInfoPoints = []
                     }
                   "More"))
             [ PFieldPun
                 SrcSpanInfo
                   { srcInfoSpan =
                       SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 14 8 15
                   , srcInfoPoints = []
                   }
                 (UnQual
                    SrcSpanInfo
                      { srcInfoSpan =
                          SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 14 8 15
                      , srcInfoPoints = []
                      }
                    (Ident
                       SrcSpanInfo
                         { srcInfoSpan =
                             SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 14 8 15
                         , srcInfoPoints = []
                         }
                       "x"))
             ])
          (PTuple
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 20 8 25
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 20 8 21
                   , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 22 8 23
                   , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 24 8 25
                   ]
               }
             Boxed
             [ PVar
                 SrcSpanInfo
                   { srcInfoSpan =
                       SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 21 8 22
                   , srcInfoPoints = []
                   }
                 (Ident
                    SrcSpanInfo
                      { srcInfoSpan =
                          SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 21 8 22
                      , srcInfoPoints = []
                      }
                    "x")
             , PWildCard
                 SrcSpanInfo
                   { srcInfoSpan =
                       SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 23 8 24
                   , srcInfoPoints = []
                   }
             ])
          (ExplicitBidirectional
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 26 12 0
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 8 26 8 31
                   , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 3 9 3
                   , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 12 1 12 0
                   ]
               }
             [ PatBind
                 SrcSpanInfo
                   { srcInfoSpan =
                       SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 3 9 24
                   , srcInfoPoints = []
                   }
                 (PApp
                    SrcSpanInfo
                      { srcInfoSpan =
                          SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 3 9 9
                      , srcInfoPoints = []
                      }
                    (UnQual
                       SrcSpanInfo
                         { srcInfoSpan =
                             SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 3 9 7
                         , srcInfoPoints = []
                         }
                       (Ident
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 3 9 7
                            , srcInfoPoints = []
                            }
                          "More"))
                    [ PVar
                        SrcSpanInfo
                          { srcInfoSpan =
                              SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 8 9 9
                          , srcInfoPoints = []
                          }
                        (Ident
                           SrcSpanInfo
                             { srcInfoSpan =
                                 SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 8 9 9
                             , srcInfoPoints = []
                             }
                           "x")
                    ])
                 (UnGuardedRhs
                    SrcSpanInfo
                      { srcInfoSpan =
                          SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 10 9 24
                      , srcInfoPoints =
                          [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 10 9 11 ]
                      }
                    (Tuple
                       SrcSpanInfo
                         { srcInfoSpan =
                             SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 12 9 24
                         , srcInfoPoints =
                             [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 12 9 13
                             , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 14 9 15
                             , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 23 9 24
                             ]
                         }
                       Boxed
                       [ Var
                           SrcSpanInfo
                             { srcInfoSpan =
                                 SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 13 9 14
                             , srcInfoPoints = []
                             }
                           (UnQual
                              SrcSpanInfo
                                { srcInfoSpan =
                                    SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 13 9 14
                                , srcInfoPoints = []
                                }
                              (Ident
                                 SrcSpanInfo
                                   { srcInfoSpan =
                                       SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 13 9 14
                                   , srcInfoPoints = []
                                   }
                                 "x"))
                       , Con
                           SrcSpanInfo
                             { srcInfoSpan =
                                 SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 16 9 23
                             , srcInfoPoints = []
                             }
                           (UnQual
                              SrcSpanInfo
                                { srcInfoSpan =
                                    SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 16 9 23
                                , srcInfoPoints = []
                                }
                              (Ident
                                 SrcSpanInfo
                                   { srcInfoSpan =
                                       SrcSpan "tests/examples/RecordPatternSynonyms.hs" 9 16 9 23
                                   , srcInfoPoints = []
                                   }
                                 "Nothing"))
                       ]))
                 Nothing
             ])
      , TypeSig
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/RecordPatternSynonyms.hs" 12 1 12 16
            , srcInfoPoints =
                [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 12 10 12 12 ]
            }
          [ Ident
              SrcSpanInfo
                { srcInfoSpan =
                    SrcSpan "tests/examples/RecordPatternSynonyms.hs" 12 1 12 9
                , srcInfoPoints = []
                }
              "selector"
          ]
          (TyCon
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/RecordPatternSynonyms.hs" 12 13 12 16
               , srcInfoPoints = []
               }
             (UnQual
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/RecordPatternSynonyms.hs" 12 13 12 16
                  , srcInfoPoints = []
                  }
                (Ident
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/RecordPatternSynonyms.hs" 12 13 12 16
                     , srcInfoPoints = []
                     }
                   "Int")))
      , PatBind
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 1 13 17
            , srcInfoPoints = []
            }
          (PVar
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 1 13 9
               , srcInfoPoints = []
               }
             (Ident
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 1 13 9
                  , srcInfoPoints = []
                  }
                "selector"))
          (UnGuardedRhs
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 10 13 17
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 10 13 11 ]
               }
             (App
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 12 13 17
                  , srcInfoPoints = []
                  }
                (Var
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 12 13 13
                     , srcInfoPoints = []
                     }
                   (UnQual
                      SrcSpanInfo
                        { srcInfoSpan =
                            SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 12 13 13
                        , srcInfoPoints = []
                        }
                      (Ident
                         SrcSpanInfo
                           { srcInfoSpan =
                               SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 12 13 13
                           , srcInfoPoints = []
                           }
                         "x")))
                (List
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 14 13 17
                     , srcInfoPoints =
                         [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 14 13 15
                         , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 16 13 17
                         ]
                     }
                   [ Lit
                       SrcSpanInfo
                         { srcInfoSpan =
                             SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 15 13 16
                         , srcInfoPoints = []
                         }
                       (Int
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/RecordPatternSynonyms.hs" 13 15 13 16
                            , srcInfoPoints = []
                            }
                          5
                          "5")
                   ])))
          Nothing
      , TypeSig
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/RecordPatternSynonyms.hs" 15 1 15 19
            , srcInfoPoints =
                [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 15 8 15 10 ]
            }
          [ Ident
              SrcSpanInfo
                { srcInfoSpan =
                    SrcSpan "tests/examples/RecordPatternSynonyms.hs" 15 1 15 7
                , srcInfoPoints = []
                }
              "update"
          ]
          (TyList
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/RecordPatternSynonyms.hs" 15 11 15 19
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 15 11 15 12
                   , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 15 18 15 19
                   ]
               }
             (TyCon
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/RecordPatternSynonyms.hs" 15 12 15 18
                  , srcInfoPoints = []
                  }
                (UnQual
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/RecordPatternSynonyms.hs" 15 12 15 18
                     , srcInfoPoints = []
                     }
                   (Ident
                      SrcSpanInfo
                        { srcInfoSpan =
                            SrcSpan "tests/examples/RecordPatternSynonyms.hs" 15 12 15 18
                        , srcInfoPoints = []
                        }
                      "String"))))
      , PatBind
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 1 16 38
            , srcInfoPoints = []
            }
          (PVar
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 1 16 7
               , srcInfoPoints = []
               }
             (Ident
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 1 16 7
                  , srcInfoPoints = []
                  }
                "update"))
          (UnGuardedRhs
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 8 16 38
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 8 16 9 ]
               }
             (RecUpdate
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 10 16 38
                  , srcInfoPoints =
                      [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 21 16 22
                      , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 37 16 38
                      ]
                  }
                (List
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 10 16 20
                     , srcInfoPoints =
                         [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 10 16 11
                         , SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 19 16 20
                         ]
                     }
                   [ Lit
                       SrcSpanInfo
                         { srcInfoSpan =
                             SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 11 16 19
                         , srcInfoPoints = []
                         }
                       (String
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 11 16 19
                            , srcInfoPoints = []
                            }
                          "String"
                          "String")
                   ])
                [ FieldUpdate
                    SrcSpanInfo
                      { srcInfoSpan =
                          SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 23 16 36
                      , srcInfoPoints =
                          [ SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 25 16 26 ]
                      }
                    (UnQual
                       SrcSpanInfo
                         { srcInfoSpan =
                             SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 23 16 24
                         , srcInfoPoints = []
                         }
                       (Ident
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 23 16 24
                            , srcInfoPoints = []
                            }
                          "x"))
                    (Lit
                       SrcSpanInfo
                         { srcInfoSpan =
                             SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 27 16 36
                         , srcInfoPoints = []
                         }
                       (String
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/RecordPatternSynonyms.hs" 16 27 16 36
                            , srcInfoPoints = []
                            }
                          "updated"
                          "updated"))
                ]))
          Nothing
      ]
  , [ Comment
        False
        (SrcSpan "tests/examples/RecordPatternSynonyms.hs" 11 1 11 12)
        " Selector"
    ]
  )
